.github/workflows/test_typing.yml

```diff
@@ -0,0 +1,25 @@
+name: Test typing
+
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron:  '0 2 * * *'
```

Comment:
> is this just to make it less annoying for contributors?

Correct! But the pre-commit ideas works, but mypy is kinda slow :-\

Reply:
yeah, but I don't think it will check the whole repo right?  I'm actually not sure here... it's possible that by using the `files` argument (to restrict the scope of the checks) we might actually be forcing it to run them on all of those files every time?

Reply:
>actually be forcing it to run them on all of those files every time? (even if the specific commit didn't touch anything in there)

I can check, but yeah in principle I think it would check them all the time.

Reply:
Is this discussion resolved? Can I merge now, or do we want to change before merge?

Reply:
I don't mind moving forward with this for now.  So I'll resolve this.  I'm not sure who is going to be monitoring these cron jobs, so there's a vague concern that we'll get regressions if we don't put this into PR tests eventually.  and I'm actually ok with the contributor burden of enforcing typing on PRs for sections of the code that we have already carefully fixed (as long as it's only those files)

Reply:
Yeah, agreed - i think putting into tests eventually for things we have fixed will be right move, will help prevent regressions. can do that in followup PRs